src/AuthContext.tsx

```diff
@@ -77,6 +77,7 @@ export const AuthProvider: FC<AuthProviderProps> = ({
   onSignIn,
   onSignOut,
   location = window.location,
+  args,
```

Comment:
We should probably be more descriptive with the naming of this argument. `args` seem to generic, would you agree?

Reply:
callbackState may be more appropriate

Reply:
you could get it with a callback instead of state like `getSigninState`

Reply:
Would it not be better to add return value for onBeforeSignIn so you can create the args object for each signin? This way, you could e.g. create different state during login.

Reply:
The goal is to give the parameter to signinRedirect

Reply:
Exactly, I have the same issue. But if row 120, the call to onBeforeSignIn (which is a method that can be implemented) could return a args object, it can be passed in the next row to signinRedirect.

Something like:

let args = {};
if(onBeforeSignIn) { 
 args = onBeforeSignIn();
}
userManager.signinRedirect(args);

Reply:
The onBeforeSignIn method is only called here so it seems like it is only used fo the autoSignIn flow. Hence, I think it might make sense to implement what it here.

Reply:
@hosseinmd Any progress with this PR?

Reply:
@hosseinmd yes, I think they implemented exactly this in: https://github.com/bjerkio/oidc-react/commit/33633557dccb59147cdb1dd6a6c577e4f2f4536d.